### PR TITLE
Tune checkpointing

### PR DESCRIPTION
**Pull request by krfricke**
Move from custom checkpointing callback to ray-supplied callbacks.

Depends on #20

### CODE REVIEW

**Comment by krfricke** [comment]
Actually you can wait to review this PR until #20 is merged

File: xgboost_ray/tests/test_tune.py [hunk 1]
```diff
@@ -59,6 +64,21 @@ def test_num_iters(self):
             all(analysis.results_df["training_iteration"] ==
                 analysis.results_df["config.num_boost_round"]))
 
+    def testCheckpointing(self):
+        analysis = tune.run(
+            tune.with_parameters(
+                self.train_func, callbacks=[TuneReportCheckpointCallback()]),
+            config=self.params,
+            resources_per_trial={
+                "cpu": 1,
```
Comment (richardliaw) [inline xgboost_ray/tests/test_tune.py:73]: Can you try testing with 2 or 3 workers in each xgboost job?

**Comment by amogkam** [comment]
I believe this PR also depends on https://github.com/ray-project/ray/pull/12572.

**Comment by krfricke** [comment]
Updated PR following the discussion on ray-project/ray#12572

With this replace mechanism, I'll likely add some unit tests to make sure this works as expected.

**Comment by krfricke** [comment]
Ready again for reviews @amogkam @richardliaw

File: xgboost_ray/session.py [hunk 2]
```diff
@@ -22,6 +22,11 @@ def put_queue(self, item):
 _session = None
 
 
+def is_xgboost_ray_actor() -> bool:
```
Comment (amogkam) [inline xgboost_ray/session.py:25]: Is this still needed?

**Review by amogkam — APPROVED** [review]
LGTM! Thanks @krfricke

File: xgboost_ray/session.py (hunk 2 above)
Comment (krfricke) [reply xgboost_ray/session.py]: Ah thanks for catching this! Removed it

**Pull request merged**